OpenRA.Mods.Common/DiscordService.cs

```diff
@@ -206,29 +190,26 @@ void UpdateParty(int players, int slots)
 			});
 		}
 
-		public void Dispose()
+		public void UpdateDetails(string details)
 		{
-			if (client != null)
-			{
-				client.Dispose();
-				client = null;
-				instance = null;
-			}
-		}
+			if (client == null)
+				return;
 
-		public static void UpdateStatus(DiscordState state, string details = null, string secret = null, int? players = null, int? slots = null)
-		{
-			Service?.SetStatus(state, details, secret, players, slots);
+			client.UpdateDetails(details);
 		}
 
-		public static void UpdatePlayers(int players, int slots)
+		public void Dispose()
 		{
-			Service?.UpdateParty(players, slots);
+			client?.Dispose();
 		}
 
-		public static void UpdateDetails(string details)
+		public static DiscordService Get(ModData modData)
 		{
-			Service?.client.UpdateDetails(details);
+			var manifest = modData.Manifest;
+			if (manifest.Contains<DiscordService>())
+				return manifest.Get<DiscordService>();
```

Comment:
why can't we cache he instance like the previous script did?

Reply:
On thinking about it, we can continue to do that. Adjusted back to that pattern.